swarm/storage/hasherstore.go

```diff
@@ -241,8 +243,12 @@ func (h *hasherStore) newDataEncryption(key encryption.Key) encryption.Encryptio
 
 func (h *hasherStore) storeChunk(ctx context.Context, ch Chunk) {
 	atomic.AddUint64(&h.nrChunks, 1)
+	h.tag.Inc(chunk.STORED) // this has to be here otherwise tests flake
```

Comment:
then tests should be amended... 
this increment needs to come after `Put` returns

Reply:
there is a bit more complexity involved in this case. this means we have to adjust all of our APIs to return a response only when all chunks have been stored (or at least after the `filestore.Wait` function has returned. this does not guaranty storage per-se). i am afraid this is a wider change than just mending a few tests. i do agree that it has to go after the `Put`, but i am not entirely sure about the repercussions of mending other parts of the codebase in order to conform.